wintasee/hooks/memoryhooks.cpp

```diff
@@ -0,0 +1,1121 @@
+{
+    debugprintf(__FUNCTION__ " called: %0x%p\n", HeapHandle);
+    return TRUE;
+}
+
+HOOKFUNC SIZE_T WINAPI MyHeapSize(HANDLE hHeap, DWORD dwFlags, LPCVOID lpMem)
+{
+    debugprintf(__FUNCTION__ "(0x%X 0x%p) called: 0x%p\n", dwFlags, lpMem, hHeap);
+    BOOL heap_locked = FALSE;
+    SIZE_T rv = 0;
+    for (auto& h : g_heaps())
+    {
+        if (h == hHeap)
+        {
+            if ((h->heap_flags & HEAP_NO_SERIALIZE) != HEAP_NO_SERIALIZE ||
+                (dwFlags & HEAP_NO_SERIALIZE) != HEAP_NO_SERIALIZE)
+            {
+                MyHeapLock(hHeap);
+                heap_locked = TRUE;
+            }
+            for (auto& s : h->heap_segments)
+            {
+                if (s == lpMem)
+                {
+                    rv = MemoryManager::GetSizeOfAllocation(s);
+                    goto heap_size_done;
+                }
+            }
+            SetLastError(ERROR_INVALID_PARAMETER);
+            goto heap_size_done;
+        }
+    }
+    SetLastError(ERROR_INVALID_HANDLE);
+heap_size_done:
+    if (heap_locked == TRUE)
+    {
+        MyHeapUnlock(hHeap);
+    }
+    return rv;
+}
+
+HOOKFUNC BOOL WINAPI MyHeapUnlock(HANDLE hHeap)
+{
+    debugprintf(__FUNCTION__ " called: 0x%p\n", hHeap);
+    for (auto& h : g_heaps())
+    {
+        if (h == hHeap)
+        {
+            h->heap_lock.clear();
+            return TRUE;
+        }
+    }
+    SetLastError(ERROR_INVALID_HANDLE);
+    return FALSE;
+}
+
+HOOKFUNC BOOL WINAPI MyHeapValidate(HANDLE hHeap, DWORD dwFlags, LPCVOID lpMem)
+{
+    debugprintf(__FUNCTION__ "(0x%X 0x%p) called: 0x%p\n", dwFlags, lpMem, hHeap);
+    BOOL heap_locked = FALSE;
+    BOOL rv = FALSE;
+    if (hHeap != nullptr)
+    {
+        for (auto &h : g_heaps())
+        {
+            if (h == hHeap)
+            {
+                if ((h->heap_flags & HEAP_NO_SERIALIZE) != HEAP_NO_SERIALIZE ||
+                    (dwFlags & HEAP_NO_SERIALIZE) != HEAP_NO_SERIALIZE)
+                {
+                    MyHeapLock(hHeap);
+                    heap_locked = TRUE;
+                }
+                for (auto& s : h->heap_segments)
+                {
+                    if (lpMem == nullptr || s == lpMem)
+                    {
+                        if (MyIsBadReadPtr(s, MemoryManager::GetSizeOfAllocation(s)) == FALSE &&
+                            MyIsBadWritePtr(s, MemoryManager::GetSizeOfAllocation(s)) == FALSE)
+                        {
+                            rv = TRUE;
+                            goto heap_validate_done;
+                        }
+                    }
+                }
+            }
+        }
+    }
+heap_validate_done:
+    if (heap_locked == TRUE)
+    {
+        MyHeapUnlock(hHeap);
+    }
+    return rv;
+}
+
+HOOKFUNC BOOL WINAPI MyHeapWalk(HANDLE hHeap, LPPROCESS_HEAP_ENTRY lpEntry)
+{
+    debugprintf(__FUNCTION__ "(0x%p) called: 0x%p\n", lpEntry, hHeap);
+    MessageBoxA(nullptr,//gamehwnd,

```

Comment:
Placeholder, because it's meant to be used, but that breaks compilation due to header issues...